integration-cli/docker_cli_run_unix_test.go

```diff
@@ -196,6 +196,75 @@ func (s *DockerSuite) TestRunAttachDetachFromFlag(c *check.C) {
 	c.Assert(running, checker.Equals, "true", check.Commentf("expected container to still be running"))
 }
 
+// TestRunDetach checks attaching and detaching with the escape sequence specified via flags.
+func (s *DockerSuite) TestRunAttachDetachFromInvalidFlag(c *check.C) {
```

Comment:
Not only. 
I can find two existing `--detach-key` tests
* TestRunAttachDetachKeysOverrideConfig: checks attaching and detaching with the detach flags, making sure it overrides config file.
* TestRunDetach: checks attaching and detaching with the detach flags, making sure it overrides config file

What this test is doing is: input an invalid detach key, it will throw a warning, then fall back to default detach keys, make sure default key works properly. 

Currently we don't have an integration test for testing default detach key as I can see, so this will also be a good supplement.